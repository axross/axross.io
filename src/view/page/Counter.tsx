import { ClassAttributes, createElement, MouseEventHandler } from 'react';

type Props = ClassAttributes<HTMLElement> & {
  count: number;
  onClickIncrement: MouseEventHandler<HTMLElement>;
  onClickDecrement: MouseEventHandler<HTMLElement>;
  className?: string;
};

const Counter = ({ count, onClickIncrement, onClickDecrement, className }: Props) => (
  <div className={className}>
    count: {count}
    <button onClick={onClickIncrement}>increment</button>
    <button onClick={onClickDecrement}>decrement</button>
  </div>
);

export default Counter;
