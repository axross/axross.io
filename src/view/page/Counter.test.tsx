import { createElement } from 'react';
import { create } from 'react-test-renderer';
import { render, Simulate } from 'react-testing-library';
import Counter from './Counter';

describe('<Counter>', () => {
  test('it matches with the previous snapshot', () => {
    const renderer = create(
      <Counter count={5} onClickIncrement={() => undefined} onClickDecrement={() => undefined} />,
    );

    expect(renderer.toJSON()).toMatchSnapshot();
  });

  test('it fires onIncrement() when the first button is clicked', () => {
    const onClickIncrement = jest.fn();
    const event = { clientX: Date.now() };

    const { getByText } = render(
      <Counter count={0} onClickIncrement={onClickIncrement} onClickDecrement={() => undefined} />,
    );
    const incrementButton = getByText('increment');

    Simulate.click(incrementButton, event);

    expect(onClickIncrement).toHaveBeenCalledWith(expect.objectContaining(event));
  });

  test('it fires onDecrement() when the second button is clicked', () => {
    const onClickDecrement = jest.fn();
    const event = { clientX: Date.now() };

    const { getByText } = render(
      <Counter count={0} onClickIncrement={() => undefined} onClickDecrement={onClickDecrement} />,
    );
    const decrementButton = getByText('decrement');

    Simulate.click(decrementButton, event);

    expect(onClickDecrement).toHaveBeenCalledWith(expect.objectContaining(event));
  });
});
