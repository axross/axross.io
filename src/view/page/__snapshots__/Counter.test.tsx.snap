// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter> it matches with the previous snapshot 1`] = `
<div>
  count: 
  5
  <button
    onClick={[Function]}
  >
    increment
  </button>
  <button
    onClick={[Function]}
  >
    decrement
  </button>
</div>
`;
